packages/kit/src/runtime/client/router.js

```diff
@@ -177,15 +179,17 @@ export class Router {
 	 * @param {string[]} chain
 	 */
 	async goto(href, { noscroll = false, replaceState = false } = {}, chain) {
-		const url = new URL(href, get_base_uri(document));
-		const selected = this.select(url);
+		if (this.client_side_routing) {
```

Comment:
It might be better to have this check in `renderer.js` instead like `if (this.client_side_routing) router.goto(...) else location.href = ...;` and avoid touching the router at all. The router is purely a client-side thing so if it's disabled it might be cleaner not to call it at all. Because I think if we were going to have this check there then we might want equivalent checks in `navigate`, etc.

Reply:
I assumed it'd be better practice to leave `goto` available to still be importable and usable even with `clientSideRouting: false`, so I did the change in the `goto `and instead didn't call `init` on the router in `start.js`

(I actually started with individual checks in `navigate`, etc, but then thought there wouldn't even a point be a point in calling `init` on the router)

Reply:
I take back what I said about `navigate`. That's not exposed to the user (I think it was in Sapper). So I guess I'm okay with this